library/core/src/main/java/com/google/android/exoplayer2/text/subrip/SubripDecoder.java

```diff
@@ -108,12 +156,122 @@ protected SubripSubtitle decode(byte[] bytes, int length, boolean reset) {
     return new SubripSubtitle(cuesArray, cueTimesUsArray);
   }
 
+  /**
+   * Process the given line by first trimming it then extracting the tags from it
+   * <p>
+   * The pattern that is used to extract the tags is specified in SSA v4+ specs and
+   * has the following form: "{\...}".
+   * <p>
+   * "All override codes appear within braces {}"
+   * "All override codes are always preceded by a backslash \"
+   *
+   * @param currentLine Current line
+   * @param tags        Extracted tags will be stored in this array list
+   * @return Processed line
+   */
+  private String processLine(String currentLine, ArrayList<String> tags) {
+    // Trim line
+    String trimmedLine = currentLine.trim();
+
+    // Extract tags
+    int replacedCharacters = 0;
+    StringBuilder processedLine = new StringBuilder(trimmedLine);
+    Matcher matcher = SUBRIP_TAG_PATTERN.matcher(trimmedLine);
+
+    while (matcher.find()) {
+      String tag = matcher.group();
+      tags.add(tag);
+      processedLine.replace(matcher.start() - replacedCharacters, matcher.end() - replacedCharacters, "");
+      replacedCharacters += tag.length();
+    }
+
+    return processedLine.toString();
+  }
+
+  /**
+   * Build a {@link Cue} based on the given text and tag
+   * <p>
+   * Match the alignment tag and calculate the line, position, position anchor accordingly
+   * <p>
+   * Based on SSA v4+ specs the alignment tag can have the following form: {\an[1-9},
+   * where the number specifies the direction (based on the numpad layout).
+   * Note. older SSA scripts may contain tags like {\a1[1-9]} but these are based on
+   * other direction rules, but multiple sources says that these are deprecated, so no support here either
+   *
+   * @param alignmentTag Alignment tag
+   * @return Built cue
+   */
+  private Cue buildCue(Spanned text, String alignmentTag) {
+    // Default values used for positioning the subtitle in case of align tags
+    float line = DEFAULT_END_FRACTION, position = DEFAULT_MID_FRACTION;
```

Comment:
No need to assign initial values to any of these (it's better not to, since that'll make the compiler ensure they're initialized along all code paths below). Add `default:` to the last case in the switch statement if needed to make the compiler happy.